src/py42/exceptions.py

```diff
@@ -255,6 +255,13 @@ def __init__(self, exception, rule_id):
         super(Py42NotFoundError, self).__init__(exception, message)
 
 
+class Py42InvalidActionError(Py42BadRequestError):
```

Comment:
This is custom error is no different than the bad request error and does virtually nothing... This is not what I meant in my other comment!

It should be something like this:

```
class Py42UpdateClosedCaseError(Py42BadRequestError):
"""An error raised when the the operation/action is not permitted."""

  def __init__(exception):
    msg = "Cannot update a closed case."
    super(Py42InvalidActionError, self).__init__(exception, msg)
```

The message is part of the init func.  See other errors in this file as an example.

Reply:
Changed. 

In this expected style of writing a new error class is created for every case. I personally feel it is too much work.

We could create more generic error name classes and respective modules can pass their appropriate messages. That is why I renamed it. `InvalidActionError` or `InvalidOperationError` can be reused in other features as well and all those module need to send is the appropriate message respective to its feature.  Its just a different style of doing the same thing.

Reply:
So say I am writing a program that does stuff with cases and handles the error for when trying to update a closed case... 

Which error makes more sense to catch?  If you just catch one that says `InvalidOperationError`, than how do you know what the invalid operation was unless you also check the error message.  By having more specific errors, it makes so that the users know the condition for which the error is raised and can handle appropriately, rather than having to figure it out on their own.

Reply:
I understand your view. Just for the sake of argument, yes `messages` convey the needed information. From my perspective the user would read the message. imo, that is the intended purpose of the messages. Here instead of conveying the information through messages we are conveying through error class names.

Reply:
You cannot (well okay, you can, but shouldn't) handle errors based on what the error message is. The point of an exception is not only to tell the user what went wrong, but also to give them a path to handling that specific error in their own code.

Reply:
I see your point and does makes sense in context when there can be multiple reasons of a failure, its better to provide different errror(classes). 
However this is very relative and differs on how we see it, in the code itself we look at the message like "EDITS_NOT_ON_CLOSED" and decide what action to take. So both ways works. I guess,  our team's preference way is  to go with new error class.

Reply:
We inspect the message because that's what we _have_ to do in order create the exception, but the aim here is to make this a better experience for users.